dask/dataframe/io/parquet.py

```diff
@@ -1107,51 +1056,62 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
     --------
     to_parquet
     """
-    is_ParquetFile = False
-    try:
-        import fastparquet
-        if isinstance(path, fastparquet.api.ParquetFile):
-            if path.open != fastparquet.util.default_open:
-                assert (re.match('.*://', path.fn)), \
-                       ("ParquetFile: Path must contain protocol" +
-                        " (e.g., s3://...) when using other than the default" +
-                        " LocalFileSystem. Path given: " + path.fn)
-
-            assert (engine in ['auto', 'fastparquet']), \
-                   ("'engine' should be set to 'auto' or 'fastparquet' " +
-                    'when reading from fastparquet.ParquetFile')
-            is_ParquetFile = True
-    except ImportError:
-            pass
-
-    if is_ParquetFile:
-        read = get_engine('fastparquet')['read']
-        if path.fn.endswith('_metadata'):
-            # remove '_metadata' from path
-            urlpath = path.fn[:-len('_metadata')]
-        else:
-            urlpath = path.fn
+    if isinstance(columns, str):
+        df = read_parquet(path, [columns], filters, categories, index,
+                          storage_options, engine, infer_divisions)
+        return df[columns]
 
-        fs, fs_token, paths = get_fs_token_paths(
-            urlpath,
-            mode='rb',
-            storage_options=storage_options
-        )
-        paths = path
+    name = 'read-parquet-' + tokenize(path, columns, filters, categories,
+                                      index, storage_options, engine,
+                                      infer_divisions)
+
+    read = get_engine(engine)['read']
+    fs, fs_token, paths = get_fs_token_paths(
+        path, mode='rb',
+        storage_options=storage_options
+    )
+
+    paths = sorted(paths, key=natural_sort_key)  # numeric rather than glob ordering
+
+    func, meta, statistics, parts = read(fs, fs_token, paths, columns=columns, filters=filters,
+                                         categories=categories, index=index,
+                                         gather_statistics=infer_divisions)
+
+    if statistics:
+        parts, statistics = zip(*[(part, stats)
+                                 for part, stats in zip(parts, statistics)
+                                 if stats['num-rows'] > 0])
+
+        # TODO: apply filters
+
+        out = sorted_columns(statistics)
+
+        if len(out) == 1:
+            divisions = out[0]['divisions']
+            index = out[0]['name']
+        elif len(out) > 1:
+            warnings.warn("Multiple sorted columns found: " + ", ".join(o['name'] for o in out))
```

Comment:
i.e., the user should ideally pick one? If they *have* picked one, then we don't care that multiple possible index columns exist. Conversely, they may have said no index, or the given index column is not sorted, so there being exactly one sorted column shouldn't mean it automatically becomes the index.

Reply:
Yes, ideally we listen to the `index=` keyword here.  I haven't gotten to this yet.